lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -304,3 +310,166 @@ def test_access_control(self):
+                "comments_count": 0,
+                "unread_comments_count": 0,
+            },
+        ]
+        expected_threads = [
+            {
+                "id": "test_thread_id_0",
+                "course_id": unicode(self.course_key),
+                "topic_id": "topic_x",
+                "created_at": "2015-04-28T00:00:00Z",
+                "updated_at": "2015-04-28T11:11:11Z",
+                "type": "discussion",
+                "title": "Test Title",
+                "raw_body": "Test body",
+                "pinned": False,
+                "closed": False,
+                "comment_count": 5,
+                "unread_comment_count": 3,
+            },
+            {
+                "id": "test_thread_id_1",
+                "course_id": unicode(self.course_key),
+                "topic_id": "topic_y",
+                "created_at": "2015-04-28T22:22:22Z",
+                "updated_at": "2015-04-28T00:33:33Z",
+                "type": "question",
+                "title": "Another Test Title",
+                "raw_body": "More content",
+                "pinned": False,
+                "closed": True,
+                "comment_count": 18,
+                "unread_comment_count": 0,
+            },
+            {
+                "id": "test_thread_id_2",
+                "course_id": unicode(self.course_key),
+                "topic_id": "topic_x",
+                "created_at": "2015-04-28T00:44:44Z",
+                "updated_at": "2015-04-28T00:55:55Z",
+                "type": "discussion",
+                "title": "Yet Another Test Title",
+                "raw_body": "Still more content",
+                "pinned": True,
+                "closed": False,
+                "comment_count": 0,
+                "unread_comment_count": 0,
+            },
+        ]
+        self.assertEqual(
+            self.get_thread_list(source_threads),
+            {
+                "results": expected_threads,
+                "next": None,
+                "previous": None,
+            }
+        )
+
+    def test_pagination(self):
+        # N.B. Empty thread list is not realistic but convenient for this test
+        self.assertEqual(
+            self.get_thread_list([], page=1, num_pages=3),
+            {
+                "results": [],
+                "next": "http://testserver/test_path?page=2",
+                "previous": None,
+            }
+        )
+        self.assertEqual(
+            self.get_thread_list([], page=2, num_pages=3),
+            {
+                "results": [],
+                "next": "http://testserver/test_path?page=3",
+                "previous": "http://testserver/test_path?page=1",
+            }
+        )
+        self.assertEqual(
+            self.get_thread_list([], page=3, num_pages=3),
+            {
+                "results": [],
+                "next": None,
+                "previous": "http://testserver/test_path?page=2",
+            }
+        )
```

Comment:
What about the case where the page requested is more than the number of available pages?